### PR TITLE
task -> thread

### PR DESCRIPTION
**Pull request by ajnirp**
for #8512 

yay for Python's `os.walk`

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8522)

### CODE REVIEW

**Comment by Manishearth** [comment]
Reviewed 42 of 149 files at r1.
Review status: 31 of 125 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[ORGANIZATION.md, line 74 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K34OhCr_dayuAlsz71A)** ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/ORGANIZATION.md#L74)):</sup>
Don't think this should have been renamed. Not sure/.


---

<sup>**[ports/cef/interfaces/cef_render_process_handler.rs, line 116 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K34OvWpCZTUhHYlnUgE)** ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/ports/cef/interfaces/cef_render_process_handler.rs#L116)):</sup>
I think "task" is CEF language, so it shouldn't be replaced. ditto for everything else in cef/x


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8522)

**Comment by wenderen** [comment]
Review status: 31 of 125 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[ORGANIZATION.md, line 74 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K34OhCr_dayuAlsz71A)** ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/ORGANIZATION.md#L74)):</sup>
i'm awaiting confirmation to roll back the renaming


---

<sup>**[ports/cef/interfaces/cef_render_process_handler.rs, line 116 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K34OvWpCZTUhHYlnUgE)** ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/ports/cef/interfaces/cef_render_process_handler.rs#L116)):</sup>
reverted


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8522)

**Comment by jdm** [comment]
Yeah, rust-task_info refers to Mach tasks (ie. the OS X kernel): https://github.com/wenderen/servo/blob/8512-rename-task-to-thread/support/rust-thread_info/src/task_info.c

**Comment by wenderen** [comment]
(ping)

**Comment by Manishearth** [comment]
Reviewed 106 of 149 files at r1.
Review status: 109 of 115 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

_[components/profile/Cargo.toml, line 32 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K3tTApj46HhU0_VrIEj:-K3tTApj46HhU0_VrIEk:-1461141721) ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/components/profile/Cargo.toml#L32)):_
did we rename this dep?xc

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8522)_

**Comment by Manishearth** [comment]
Reviewed 4 of 13 files at r2, 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8522)_

**Comment by wenderen** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

_[components/profile/Cargo.toml, line 32 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8522#-K3tTApj46HhU0_VrIEj:-K3tvyGCPxa07c_5bFE3:677916264) ([raw file](https://github.com/servo/servo/blob/036787a3ecf76c17c2604c497f5ede2cf91b0c88/components/profile/Cargo.toml#L32)):_
oops

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8522)_

File: components/util/workqueue.rs [hunk 1]
```diff
@@ -300,7 +300,7 @@ impl<QueueData: Sync, WorkData: Send> WorkQueue<QueueData, WorkData> {
         self.work_count += 1
     }
 
-    /// Synchronously runs all the enqueued tasks and waits for them to complete.
+    /// Synchronously runs all the enqueued threads and waits for them to complete.
```
Comment (jdm) [inline components/util/workqueue.rs]: This should be reverted.

File: ports/cef/interfaces/cef_task.rs [hunk 2]
```diff
@@ -47,15 +47,15 @@ use std::mem;
 use std::ptr;
 
 //
-// Implement this structure for asynchronous task execution. If the task is
+// Implement this structure for asynchronous thread execution. If the thread is
```
Comment (jdm) [inline ports/cef/interfaces/cef_task.rs]: The changes to this file will need to be reverted, since it's a CEF API.

File: components/script/dom/websocket.rs [hunk 3]
```diff
@@ -466,13 +466,13 @@ impl WebSocketMethods for WebSocket {
 }
 
 
-/// Task queued when *the WebSocket connection is established*.
-struct ConnectionEstablishedTask {
+/// Thread queued when *the WebSocket connection is established*.
```
Comment (jdm) [inline components/script/dom/websocket.rs]: The changes for this type and comment should be reverted.

File: components/script/dom/websocket.rs [hunk 4]
```diff
@@ -496,11 +496,11 @@ impl Runnable for ConnectionEstablishedTask {
     }
 }
 
-struct BufferedAmountTask {
+struct BufferedAmountThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Please revert this change.

File: components/script/dom/websocket.rs [hunk 5]
```diff
@@ -514,11 +514,11 @@ impl Runnable for BufferedAmountTask {
     }
 }
 
-struct CloseTask {
+struct CloseThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Please revert this.

File: components/script/dom/websocket.rs [hunk 6]
```diff
@@ -551,16 +551,16 @@ impl Runnable for CloseTask {
     }
 }
 
-struct MessageReceivedTask {
+struct MessageReceivedThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Please revert this.

File: components/script/dom/filereader.rs [hunk 7]
```diff
@@ -404,27 +404,27 @@ impl Runnable for FileReaderEvent {
...
 
-// https://w3c.github.io/FileAPI/#task-read-operation
+// https://w3c.github.io/FileAPI/#thread-read-operation
 fn perform_annotated_read_operation(gen_id: GenerationId, data: ReadMetaData, blob_contents: Receiver<Vec<u8>>,
     filereader: TrustedFileReader, script_chan: Box<ScriptChan + Send>) {
     let chan = &script_chan;
     // Step 4
-    let task = box FileReaderEvent::ProcessRead(filereader.clone(), gen_id);
-    chan.send(CommonScriptMsg::RunnableMsg(FileRead, task)).unwrap();
+    let thread = box FileReaderEvent::ProcessRead(filereader.clone(), gen_id);
```
Comment (jdm) [inline components/script/dom/filereader.rs]: This rename should be reverted.

File: components/script/dom/filereader.rs [hunk 8]
```diff
@@ -404,27 +404,27 @@ impl Runnable for FileReaderEvent {
     }
 }
 
-// https://w3c.github.io/FileAPI/#task-read-operation
+// https://w3c.github.io/FileAPI/#thread-read-operation
```
Comment (jdm) [inline components/script/dom/filereader.rs]: This needs to be reverted.

File: components/script/dom/filereader.rs [hunk 9]
```diff
@@ -404,27 +404,27 @@ impl Runnable for FileReaderEvent {
...
     let chan = &script_chan;
     // Step 4
-    let task = box FileReaderEvent::ProcessRead(filereader.clone(), gen_id);
-    chan.send(CommonScriptMsg::RunnableMsg(FileRead, task)).unwrap();
+    let thread = box FileReaderEvent::ProcessRead(filereader.clone(), gen_id);
+    chan.send(CommonScriptMsg::RunnableMsg(FileRead, thread)).unwrap();
 
-    let task = box FileReaderEvent::ProcessReadData(filereader.clone(), gen_id);
-    chan.send(CommonScriptMsg::RunnableMsg(FileRead, task)).unwrap();
+    let thread = box FileReaderEvent::ProcessReadData(filereader.clone(), gen_id);
```
Comment (jdm) [inline components/script/dom/filereader.rs]: Revert.

File: components/script/dom/filereader.rs [hunk 10]
```diff
@@ -404,27 +404,27 @@ impl Runnable for FileReaderEvent {
...
+    chan.send(CommonScriptMsg::RunnableMsg(FileRead, thread)).unwrap();
 
     let bytes = match blob_contents.recv() {
         Ok(bytes) => bytes,
         Err(_) => {
-            let task = box FileReaderEvent::ProcessReadError(filereader,
+            let thread = box FileReaderEvent::ProcessReadError(filereader,
```
Comment (jdm) [inline components/script/dom/filereader.rs]: Revert.

File: components/script/dom/filereader.rs [hunk 11]
```diff
@@ -404,27 +404,27 @@ impl Runnable for FileReaderEvent {
...
                 gen_id, DOMErrorName::NotFoundError);
-            chan.send(CommonScriptMsg::RunnableMsg(FileRead, task)).unwrap();
+            chan.send(CommonScriptMsg::RunnableMsg(FileRead, thread)).unwrap();
             return;
         }
     };
 
-    let task = box FileReaderEvent::ProcessReadEOF(filereader, gen_id, data, bytes);
-    chan.send(CommonScriptMsg::RunnableMsg(FileRead, task)).unwrap();
+    let thread = box FileReaderEvent::ProcessReadEOF(filereader, gen_id, data, bytes);
```
Comment (jdm) [inline components/script/dom/filereader.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 12]
```diff
@@ -248,20 +248,20 @@ impl WebSocket {
                 Ok(channel) => channel,
                 Err(e) => {
                     debug!("Failed to establish a WebSocket connection: {:?}", e);
-                    let task = box CloseTask {
+                    let thread = box CloseThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 13]
```diff
@@ -248,20 +248,20 @@ impl WebSocket {
...
                     };
-                    sender.send(CommonScriptMsg::RunnableMsg(WebSocketEvent, task)).unwrap();
+                    sender.send(CommonScriptMsg::RunnableMsg(WebSocketEvent, thread)).unwrap();
                     return;
                 }
             };
             let ws_sender = Arc::new(Mutex::new(ws_sender));
 
-            let open_task = box ConnectionEstablishedTask {
+            let open_thread = box ConnectionEstablishedThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 14]
```diff
@@ -274,19 +274,19 @@ impl WebSocket {
                     Ok(Message::Pong(_)) => continue,
                     Ok(Message::Close(data)) => {
                         ws_sender.lock().unwrap().send_message(Message::Close(data)).unwrap();
-                        let task = box CloseTask {
+                        let thread = box CloseThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 15]
```diff
@@ -274,19 +274,19 @@ impl WebSocket {
...
                             addr: address,
                         };
-                        sender.send(CommonScriptMsg::RunnableMsg(WebSocketEvent, task)).unwrap();
+                        sender.send(CommonScriptMsg::RunnableMsg(WebSocketEvent, thread)).unwrap();
                         break;
                     },
                     Err(_) => break,
                 };
-                let message_task = box MessageReceivedTask {
+                let message_thread = box MessageReceivedThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 16]
```diff
@@ -327,11 +327,11 @@ impl WebSocket {
         if !self.clearing_buffer.get() && self.ready_state.get() == WebSocketRequestState::Open {
             self.clearing_buffer.set(true);
 
-            let task = box BufferedAmountTask {
+            let thread = box BufferedAmountThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

File: components/script/dom/websocket.rs [hunk 17]
```diff
@@ -551,16 +551,16 @@ impl Runnable for CloseTask {
...
 }
 
-struct MessageReceivedTask {
+struct MessageReceivedThread {
     address: Trusted<WebSocket>,
     message: MessageData,
 }
 
-impl Runnable for MessageReceivedTask {
+impl Runnable for MessageReceivedThread {
```
Comment (jdm) [inline components/script/dom/websocket.rs]: Revert.

**Comment by jdm** [comment]
@bors-servo: delegate+
With those changes and a rebase this should be good to land.

**Comment by KiChjang** [comment]
@wenderen What's the status of this?

**Comment by wenderen** [comment]
@KiChjang apologies for the late response, I was on vacation until yesterday.

The rebase is halfway complete (I abandoned the old rebase and re-fetched, so some of my former progress was lost). Should be done by tomorrow night.

**Comment by wenderen** [comment]
rebased, squashed and pushed

**Comment by jdm** [comment]
@wenderen There are still some incorrect renames that I identified that require addressing, if you've got the time.

**Comment by wenderen** [comment]
Sure, which renames?

**Comment by jdm** [comment]
@wenderen See all the comments on the earlier comments further up the page. The ones that are still visible haven't been addressed.

**Comment by wenderen** [comment]
Oops. How do I  reopen this?

**Comment by frewsxcv** [comment]
You might have deleted the branch. It'd probably be easier just to open a new pr

**Comment by wenderen** [comment]
@frewsxcv done: #9201
